src/main/java/cinema/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package cinema.service;
+
+import cinema.dao.OrderDao;
+import cinema.lib.Inject;
+import cinema.lib.ServiceImpl;
+import cinema.model.Order;
+import cinema.model.ShoppingCart;
+import cinema.model.User;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@ServiceImpl
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private ShoppingCartService shoppingCartService;
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setTickets(shoppingCart.getTickets());
+        order.setUser(shoppingCart.getUser());
+        orderDao.add(order);
+        shoppingCartService.clear(shoppingCart);
```

Comment:
if it is like this - 26 line is redundant. Shopping cart's tickets getting cleared on 23 line. make a copy of tickets list and it'll be ok